backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentService.java

```diff
@@ -31,21 +30,43 @@
     @Autowired
     private ModelMapper modelMapper;
 
-    public EquipmentResponseDTO getEquipment(String code){
+    public EquipmentResponseDTO getEquipment(String code) {
         Equipment equipment = equipmentRepository.findByCode(code);
-        if(equipment == null){
+        if (equipment == null) {
             throw new CustomException("No such an equipment found", HttpStatus.BAD_REQUEST);
         }
         List<HistoricalValue> history = historicalValueRepository.findAllByEquipment_Code(code);
         history.sort((o1, o2) -> o1.getTimestamp().compareTo(o2.getTimestamp()));
-        List<EquipmentHistoryDTO> hist = history.stream().map(h ->
-                (modelMapper.map(h,EquipmentHistoryDTO.class))).collect(Collectors.toList());
+        List<EquipmentHistoryDTO> hist = history.stream().map(h -> (modelMapper.map(h, EquipmentHistoryDTO.class)))
+                .collect(Collectors.toList());
+
+        return new EquipmentResponseDTO(equipment, hist);
+    }
+
+    /**
+     * Calculates the change ratio of the specified currency, respected to current
+     * value and yesterday's closing
+     * 
+     * @param code
+     * @return change ratio (in percent)
+     */
+    public double getDailyChange(String code) {
+
+        EquipmentResponseDTO equipment = this.getEquipment(code);
+
+        List<EquipmentHistoryDTO> hist = equipment.getHistoricalValues();
+
+        int histSize = hist.size();
+
+        double currentVal = equipment.getEquipment().getCurrentValue();
```

Comment:
history might not be initialized on startup. We mush add a handler here if `hist.size == 0`